### PR TITLE
Add an ignoreProtected parameter for removing and adding members to grou...

### PR DESCRIPTION
**Pull request by colinschoen**
...ps. Signed off by Colin Schoen colin@simplemachines.org

### CODE REVIEW

File: Sources/Subs-Membergroups.php [hunk 1]
```diff
@@ -312,7 +313,7 @@ function removeMembersFromGroups($members, $groups = null, $permissionCheckDone
 	$groups = array_diff($groups, $implicitGroups);
 
 	// Don't forget the protected groups.
-	if (!allowedTo('admin_forum'))
+	if (!allowedTo('admin_forum') && !ignoreProtected)
```
Comment (eurich) [inline Sources/Subs-Membergroups.php]: !ignoreProtected would validate a constant. Haven't checked the details but I'd expect a variable check like !empty($ignoreProtected). You could also set $ignoreProtected = null and do an !isset($ignoreProtected) instead.

File: Sources/Subs-Membergroups.php [hunk 2]
```diff
@@ -487,7 +489,7 @@ function addMembersToGroup($members, $group, $type = 'auto', $permissionCheckDon
 	if (!allowedTo('admin_forum') && $group == 1)
 		return false;
 	// ... and assign protected groups!
-	elseif (!allowedTo('admin_forum'))
+	elseif (!allowedTo('admin_forum') && !ignoreProtected)
```
Comment (eurich) [inline Sources/Subs-Membergroups.php]: same here..

File: Sources/Subs-Membergroups.php (hunk 1 above)
Comment (colinschoen) [inline Sources/Subs-Membergroups.php]: I could change it to a constant, good idea. I will see if anyone else has any comments otherwise I will go for it.

Comment (emanuele45) [inline Sources/Subs-Membergroups.php]: Why would you make it a constant?
I think eurich wanted to say that you forgot the $ in front of ignoreProtected, that's all. ;)

Comment (colinschoen) [inline Sources/Subs-Membergroups.php]: Woops, I didn't mean make a constant I meant change it from a boolean, but I suppose that is fine as it is in line $permissionCheckDone. Thanks, I didn't notice I forgot the $. I will update that. Cheers Ema.

**Comment by colinschoen** [comment]
That should fix it.

**Comment by colinschoen** [comment]
Bump

**Pull request merged**